spec/puppet/resource_api_spec.rb

```diff
@@ -408,105 +408,308 @@ def set(_context, changes)
     before(:each) do
       stub_const('Puppet::Provider::Canonicalizer', Module.new)
       stub_const('Puppet::Provider::Canonicalizer::Canonicalizer', provider_class)
+      Puppet.settings[:strict] = :off
```

Comment:
Set a global default using https://relishapp.com/rspec/rspec-core/docs/hooks/before-and-after-hooks#define-%60before%60-and-%60after%60-blocks-in-configuration . That way we don't have to worry about puppet changing under our feet across versions, and it becomes clearer if and when individual tests have to override.

And the default should be `:error`, because we need to be able to work under the strictest setting without crashing and burning.